### ISSUE TITLE
Server 2.0.0 issues - scanning libraries - socket disconnects - 'typeError'

### ISSUE THREAD

**Issue by FlamingoDaBird** [advplyr/audiobookshelf#470]
Hi.
I recently upgraded my unraid with the docker server version 2.0.0. The android app 0.9.40-beta stopped working (empty libraries) the server version showed all libraries  so no issues there.

I did a new install (renamed the app folder for audiobookshelf on my unraid to get a fresh install) added the libraries again, then I got some weird issues and error messages and the servers disconnects and gets unresponsive, the docker images stops sometimes or requires a restart.

from the docker error log:

______________________________________________________________

2022-04-20T20:12:07.310Z] INFO: === Starting Server ===
[2022-04-20T20:12:07.311Z] INFO: [Server] Init v2.0.0
[2022-04-20T20:12:07.329Z] INFO: [DB] 0 Library Items Loaded
[2022-04-20T20:12:07.329Z] INFO: [DB] 0 Collections Loaded
[2022-04-20T20:12:07.329Z] INFO: [DB] 0 Authors Loaded
[2022-04-20T20:12:07.329Z] INFO: [DB] 0 Series Loaded
[2022-04-20T20:12:07.345Z] INFO: [DB] 3 Users Loaded
[2022-04-20T20:12:07.345Z] INFO: [DB] 3 Libraries Loaded
[2022-04-20T20:12:07.345Z] INFO: [Db] Root user exists, pw: true
[2022-04-20T20:12:07.347Z] INFO: [BackupManager] 0 Backups Found
[2022-04-20T20:12:07.347Z] INFO: [BackupManager] Auto Backups are disabled
[2022-04-20T20:12:07.348Z] INFO: [LogManager] Init current daily log filename: 2022-04-20.txt
[2022-04-20T20:12:07.351Z] INFO: [Watcher] Initializing watcher for "Audiobooks".
[2022-04-20T20:12:07.352Z] INFO: [Watcher] Initializing watcher for "eBooks".
[2022-04-20T20:12:07.352Z] INFO: [Watcher] Initializing watcher for "Main".
[2022-04-20T20:12:07.360Z] INFO: Listening on http://0.0.0.0:80
[2022-04-20T20:12:27.273Z] INFO: [Server] Socket Connected 9XdH2SWhXM7TZrZaAAAB
[2022-04-20T20:12:56.931Z] INFO: [Scanner] Starting library scan lscan_gipq05yozxlcatnsxv for Audiobooks
/server/njodb/index.js:103
throw error;

TypeError: Cannot read properties of null (reading 'toLowerCase')
at /server/utils/abmetadataGenerator.js:324:111
at Array.some (<anonymous>)
at /server/utils/abmetadataGenerator.js:324:64
at Array.filter (<anonymous>)
at checkUpdatedBookAuthors (/server/utils/abmetadataGenerator.js:324:32)
at Object.parseAndCheckForUpdates (/server/utils/abmetadataGenerator.js:396:35)
at Book.syncMetadataFiles (/server/objects/mediaTypes/Book.js:222:51)
at async LibraryItem.syncFiles (/server/objects/LibraryItem.js:437:11)
at async Scanner.scanNewLibraryItem (/server/scanner/Scanner.js:389:5)
at async Promise.all (index 0)

**Comment by advplyr** [comment]
still waiting on google to approve the latest android app update to work with v2. that's why i haven't pushed an official release yet.

**Comment by FlamingoDaBird** [comment]
Right... google... lets wait then. 
Thanks for the information and the time you invest within the project.

By the way is there any way to know/check  before upgrading a docker image to see if it requires a different app-version.
So that one can wait to upgrade the image until the app is released. I might have missed some information.. 
If not that's totally understandable. This is free and a work in progress! Thanks mate! :)

**Comment by advplyr** [comment]
No this shouldn't happen again. Google must have flagged the app with some issue and it got elevated which is why it is taking so long. Apple approved it right away.

**Comment by advplyr** [comment]
You can uninstall the app and install the APK directly though. https://github.com/advplyr/audiobookshelf-app/releases/tag/v0.9.41-beta
I'm still working on the ability to link your local downloaded audiobooks to audiobooks on your server, but even without that you should be able to view and play your downloaded audiobooks.
The new android app release doesnt require a server anymore to use.

**Comment by BillThyCat** [comment]
I was having similar issues, but even thru the web interface the docker instance kept crashing, especially when trying to upload new books.  Was also having the issue where the app would show no books, then a few books, then no books, then back to the socket errors.

Was it the app connecting in the background that was causing it to crash? or was there a bug of some sort (at least running under Unraid)?  I'm running on 1.7 but I really did like the visual improvements of 2.0, so hoping it gets resolved soon.

Thanks for the hard work.  Can the new version of the apk work with v 1.7 of the docker app?

BTC

**Comment by advplyr** [comment]
No the new app version doesn't work with 1.7. If you are going to install the APK directly then you will need to update the server. Otherwise I recommend just waiting until Google approves it on the play store.

**Comment by advplyr** [comment]
Also please note that if you update the server to v2 then do not downgrade because v2 is running a script that is migrating all your data to the new data model.
Downgrading from v2 to v1.7 will only create more issues.

**Comment by BillThyCat** [comment]
> Also please note that if you update the server to v2 then do not downgrade because v2 is running a script that is migrating all your data to the new data model. Downgrading from v2 to v1.7 will only create more issues.

I ended up rebuilding the whole thing from scratch when I rolled back.

**Comment by FlamingoDaBird** [comment]
Also had the issues BillThyCat mentioned.
But nabbed your android apk version and rolled back my appdata for migrated 2.0.0 folder and now every thing works fine again!  I still have my library intact.
Thanks.

**Comment by advplyr** [comment]
Ah great, thanks for letting me know.

**Comment by FlamingoDaBird** [comment]
have some issues with the new app version, I'll create a new ticket.
regarding the offline folder / downloads,

**Comment by Albuca** [comment]
Good news -- the playstore has the latest app version available. 

Version is listed as 0.9.41-beta and changelog indicates need for v2.0.0 server or greater.

**Comment by advplyr** [comment]
Release notes for [v2.0.1](https://github.com/advplyr/audiobookshelf/releases/tag/v2.0.1) should clear this up. Thanks all
